pkg/controller/mcmhub/mcmhub_controller.go

```diff
@@ -592,6 +585,21 @@ func (r *ReconcileSubscription) Reconcile(ctx context.Context, request reconcile
 			return reconcile.Result{}, nil
 		}
 
+		// Skip the reconcile if the placement decision list is not ready
+		isReadyPlacementDecision, err := r.IsReadyPlacementDecisionList(instance)
```

Comment:
I am not too comfortable with this change because it impacts the all subscriptions.
I will prefer if we only do this if the AppSub has pre/post-hooks so we don't cause any potential regressions.

Reply:
move the cluster decision check to the appsub with pre/post-hooks only